core/appmodule/module.go

```diff
@@ -12,3 +16,23 @@ type AppModule interface {
 	// IsAppModule is a dummy method to tag a struct as implementing an AppModule.
 	IsAppModule()
 }
+
+// HasBeginBlocker is the extension interface that modules should implement to run
+// custom logic before transaction processing in a block.
+type HasBeginBlocker interface {
```

Comment:
👍🏾 I prefer this name instead of https://pkg.go.dev/github.com/cosmos/cosmos-sdk@v0.46.7/types/module#BeginBlockAppModule